### DISCUSSION TITLE
Allow record and struct/class to be specified in the opposite order

### DISCUSSION THREAD

**Discussion by DavidArno** [dotnet/csharplang#5318] [General]
Currently, if I want to create a class record, I can specify it as:
```c#
record R();
```
But if I want to qualify it as a class record, or specify a struct record, I have to put that modifier after the record:
```c#
record struct R1();
record class R2();
```
The other modifiers though come before `record`:
```c#
internal readonly partial record struct R();
```
I would like to be able to put `struct` and `class` before `record` as they are - to my mind at least - modifiers of the record type, rather than record being a modifier to the class or struct. So I'd like a future version of the compiler to support any order for `struct/class` and `record`.

**Comment by HaloFour** [comment]
IIRC with the addition of struct records it was seen that `record` was a modifier to the type and not vice versa, although that makes little sense to me that you can omit `class` for `record class`.

**Reply by alrz** [reply]
There is no crying emoji reaction.
